### ISSUE TITLE
False positive in unreachable_code rule

### ISSUE THREAD

**Issue by jan-jezek** [abaplint/abaplint#3125]
```abap
IF foo( ).
  " do something

  TEST-SEAM bar.
    RETURN.
  END-TEST-SEAM.
ENDIF.

" more code
```

I know, I know, TEST-SEAM should not be used, but nevertheless the unreachable_code rule should not fire in this case as the syntax is correct.

![image](https://github.com/abaplint/abaplint/assets/17530797/050638e5-43d7-409f-8dd6-2f3538c9e835)

**Comment by jan-jezek** [comment]
Fixed, thanks 👍🏻
